Source/Foundation/bsfCore/CMakeSources.cmake

```diff
@@ -621,16 +621,31 @@ set(BS_CORE_INC_PLATFORM_LINUX
 )
 
 set(BS_CORE_SRC_PLATFORM_LINUX
+	# Use X11
 	"bsfCore/Private/Linux/BsLinuxPlatform.cpp"
 	"bsfCore/Private/Linux/BsLinuxWindow.cpp"
+	"bsfCore/Private/Linux/BsLinuxMouse.cpp"
+	"bsfCore/Private/Linux/BsLinuxKeyboard.cpp"
+
+	# Is special
 	"bsfCore/Private/Linux/BsLinuxDropTarget.cpp"
-	"bsfCore/Private/Linux/BsLinuxFolderMonitor.cpp"
+
+	# Don't use X11
 	"bsfCore/Private/Linux/BsLinuxInput.cpp"
 	"bsfCore/Private/Linux/BsLinuxGamepad.cpp"
-	"bsfCore/Private/Linux/BsLinuxMouse.cpp"
-	"bsfCore/Private/Linux/BsLinuxKeyboard.cpp"
+	"bsfCore/Private/Linux/BsLinuxFolderMonitor.cpp"
 )
 
+# Force unity builds to isolate these files into their own section. 
+# COTIRE_START_NEW_UNITY_SOURCE causes the source file to be the first one included in a new file. 
+# This isolates the X11 headers which define inane things.
+set_source_files_properties ("bsfCore/Private/Linux/BsLinuxPlatform.cpp" PROPERTIES
```

Comment:
Could this be something we do for Windows as well, to avoid the issues with "windows.h"? If so could we just automate the process and have cmake iterate over `BS_CORE_SRC_PLATFORM_LINUX` (and other platform-specific) files and ignore them all? I don't mind those files being excluded from the unity builds (and I'm assuming their headers are also excluded from PCH), if it means for a cleaner code without duplication.

Reply:
Yes, I haven't tested unity builds on windows yet as I didn't get as much time on my desktop as I had hoped. 

The reason I am hesitant to just mark this so cotire ignores it is that it would illuminate some of the purpose of the unity builds. By keeping all the X11 comparable headers in one place they get a single unity build together causing at least some level of unity-ness.

Reply:
I'm worried about things breaking in weird ways for someone unfamiliar with the codebase. 

i.e. just including a X11 header in some other Linux specific file will break the build (I'm assuming) - and likely in not a very obvious way. I don't like having such hidden requirements in the codebase, if there's a way to avoid them.

Reply:
I'm torn, as the X11 header is possibly the worst header I've ever seen besides <windows.h> so I want to shun it's usage as much as possible, but I agree that we shouldn't leave the build breaking randomly. 

I think I will let these ones out of the unity build.